provider/maas/environ.go

```diff
@@ -948,6 +970,18 @@ func (environ *maasEnviron) StartInstance(args environs.StartInstanceParams) (
 			return nil, errors.Annotatef(err, "cannot run instances")
 		}
 	}
+	done := make(chan struct{})
+
+	if args.StatusCallback != nil {
+		// the purpose of this goroutine is to provide information while the instance
+		// is coming up, once StartInstance exits instancepoller will take care but
+		// long start times can lead to confusing status stalling, there is no
+		// consequence to this goroutine never running since it means the function
+		// ended fast and instancepoller is taking care now.
+		defer close(done)
```

Comment:
Please also wait for the goroutine to stop before returning from StartInstance. We shouldn't assume that args.StatusCallback is valid to use once this method has returned.